lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -4564,6 +4573,116 @@ void glow::convertQuantizedConstants(Function *F, CompilationContext &cctx) {
+
+        for (size_t i = 0, e = node.getNumResults(); i < e; i++) {
+          if (node.getType(i)->isQuantizedType()) {
+            outType =
+                F->getParent()->uniqueType(ElemKind::Float16Ty, node.dims(i));
+          } else {
+            outType = node.getType(i);
+          }
+          outputTypes.push_back(outType);
+        }
+
+        for (size_t i = 0, e = node.getNumInputs(); i < e; i++) {
+          if (node.getNthInput(i).getType()->isQuantizedType()) {
+            inpType = F->getParent()->uniqueType(ElemKind::Float16Ty,
+                                                 node.getNthInput(i).dims());
+          } else {
+            inpType = node.getNthInput(i).getType();
+          }
+          inputTypes.push_back(inpType);
+        }
+
+        NodeInfo fp16node = NodeInfo(node.getKind(), inputTypes, outputTypes);
+        isFP16Supported = B.isOpSupported(fp16node);
+      }
+
+      // Create a Cloned Node
+      Node *clonedNode = node.clone();
+      TypeRef newTy = nullptr;
+      for (size_t i = 0, e = node.getNumResults(); i < e; i++) {
+        if (node.getType(i)->isQuantizedType()) {
+          if (isFP16Supported) {
+            newTy =
+                F->getParent()->uniqueType(ElemKind::Float16Ty, node.dims(i));
+          } else {
+            newTy = F->getParent()->uniqueType(ElemKind::FloatTy, node.dims(i));
+          }
+        } else {
+          newTy = node.getType(i);
+        }
+        clonedNode->setType(i, newTy);
+      }
+      F->addNode(clonedNode);
+
+      // Add input dequantize node
+      for (size_t i = 0, e = node.getNumInputs(); i < e; i++) {
+        auto input = node.getNthInput(i);
+        TypeRef inTy = nullptr;
+        if (input.getType()->isQuantizedType()) {
+          if (isFP16Supported) {
+            inTy = F->getParent()->uniqueType(ElemKind::Float16Ty,
+                                              input.getType()->dims());
+          } else {
+            inTy = F->getParent()->uniqueType(ElemKind::FloatTy,
+                                              input.getType()->dims());
+          }
+          std::string deqName =
+              node.getName().str() + "_dequantize" + std::to_string(i);
+          auto *DQ = F->createDequantize(deqName, input, inTy);
+          clonedNode->setNthInput(i, DQ);
+        } else {
+          clonedNode->setNthInput(i, input);
+        }
+      }
+
+      // Add output quantize node
+      unsigned numres = clonedNode->getNumResults();
+      for (size_t i = 0, e = numres; i < e; i++) {
+        auto result = clonedNode->getNthResult(i);
+        if (result.getType()->isFPType()) {
+          std::string quantName =
+              node.getName().str() + "_quantize" + std::to_string(i);
+          auto *Q = F->createQuantize(quantName, result, node.getType(i));
+          node.getNthResult(i).replaceAllUsesOfWith(Q);
+        } else {
+          node.getNthResult(i).replaceAllUsesOfWith(result);
+        }
+      }
```

Comment:
I think this is the wrong way to do this. If a hypothetical Node has one float result and another quantized result, and you changed the quantized result to float, then you'd end up creating a broken QuantizeNode that "quantizes" to float for the original float output. You should base this on the results of the original node:
```suggestion
      unsigned numres = clonedNode->getNumResults();
      for (size_t i = 0, e = numres; i < e; i++) {
        auto origResult = node->getNthResult(i);
        if (origResult.getType()->isFPType()) {
          continue;
        }
        const std::string quantName =
              node.getName().str() + "_quantize" + std::to_string(i);
        NodeValue result = clonedNode->getNthResult(i);
        auto *Q = F->createQuantize(quantName, result, node.getType(i));
        origResult.replaceAllUsesOfWith(Q);
      }
```

Reply:
I agree with you. I will make the suggested modifications. Thanks